logstash-core/lib/logstash/pipeline.rb

```diff
@@ -56,6 +56,7 @@ def self.validate_config(config_str, settings = {})
     begin
       # There should be a better way to test this
       self.new(config_str, settings)
+      true
     rescue => e
       e.message
```

Comment:
this is for sure a more clean solution, I try to keep comments on #4944 as basically is the same code ported here.